### PR TITLE
Added the account API

### PR DESCRIPTION
**Pull request by daminisatya**

### CODE REVIEW

File: loklak.php [hunk 1]
```diff
@@ -186,4 +186,49 @@ public function suggest($query="", $count=null, $order=null, $orderby=null, $sin
...
+			$error = "Something went wrong, looks like the query is wrong.";
+			$request['error'] = array_push($request, $error);
+			return json_encode($request, true);
+		}
+	}
 }
+
+
+
+
```
Comment (sudheesh001) [inline loklak.php:234]: Please Remove these trailing white lines.

File: loklak.php [hunk 2]
```diff
@@ -150,15 +150,15 @@ public function settings() {
...
+	public function suggest($query, $count=null, $order=null, $orderby=null, $since=null, $until=null) {
 		$this->requestURL = $this->baseUrl . '/api/suggest.json';
 		$this->query = $query;
 		$this->count = $count;
 		$this->order = $order;
 		$this->orderby = $orderby;
 		$this->since = $since;
 		$this->until = $until;
-		if ($query) {
+		if($query) {
```
Comment (sudheesh001) [inline loklak.php:161]: Why was this recorrected ? It was better with `if ($query)`

File: loklak.php [hunk 3]
```diff
@@ -186,4 +186,49 @@ public function suggest($query="", $count=null, $order=null, $orderby=null, $sin
 			return json_encode($request, true);
 		}
 	}
+
+	public function account($name, $action=null, $data=null) {
+		$this->requestURL = 'http://localhost:9000/api/settings.json';
```
Comment (sudheesh001) [inline loklak.php:191]: the url is `/api/account.json`

File: loklak.php [hunk 4]
```diff
@@ -186,4 +186,49 @@ public function suggest($query="", $count=null, $order=null, $orderby=null, $sin
...
+		$this->name = $name;
+		$this->action = $action;
+		$this->data = $data;
```
Comment (sudheesh001) [inline loklak.php:194]: Write a comment here that the `$data` to be sent to the function should be stringified `json`

File: loklak.php [hunk 5]
```diff
@@ -186,4 +186,49 @@ public function suggest($query="", $count=null, $order=null, $orderby=null, $sin
...
+		$headers = array();
+		$headers['User-Agent'] = "Mozilla/5.0 (Android 4.4; Mobile; rv:41.0) Gecko/41.0 Firefox/41.0";
+        $headers['From'] = "info@loklak.org";
+		if ($name) {
+			$params = array('screen_name'=>$this->name);
+			$request = Requests::request($this->requestURL, array('Accept' => 'application.json'), $params, $headers);
```
Comment (sopankhosla) [inline loklak.php:200]: Shouldn't `headers` be added before `param`?

File: loklak.php [hunk 6]
```diff
@@ -186,4 +186,49 @@ public function suggest($query="", $count=null, $order=null, $orderby=null, $sin
...
 	}
+
+	public function account($name, $action=null, $data=null) {
+		$this->requestURL = 'http://localhost:9000/api/settings.json';
+		$this->name = $name;
+		$this->action = $action;
+		$this->data = $data;
+		$headers = array();
+		$headers['User-Agent'] = "Mozilla/5.0 (Android 4.4; Mobile; rv:41.0) Gecko/41.0 Firefox/41.0";
+        $headers['From'] = "info@loklak.org";
```
Comment (sopankhosla) [inline loklak.php:197]: Indentation issues

**Comment by daminisatya** [comment]
@sudheesh001 @sopu The patch is on hold. Will get back to it after exams. Thank you!

**Comment by sudheesh001** [comment]
@daminisatya latest merge, you might have to rebase 😄

**Comment by daminisatya** [comment]
@sudheesh001 Apologies for not updating the PR. Will do it soon after my exams.

**Pull request closed without merging**